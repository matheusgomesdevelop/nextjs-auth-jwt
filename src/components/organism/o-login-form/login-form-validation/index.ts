import { initialValues, validationSchema } from './login-form-validation';

// SendForm
import { handleLoginForm } from './login-form-send';

export { initialValues, validationSchema, handleLoginForm };
