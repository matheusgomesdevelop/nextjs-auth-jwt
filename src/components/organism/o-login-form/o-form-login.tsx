import { FormEventHandler } from "react";

// Atoms
import { A_Button } from "@/components/atoms";

// Molecules
import { M_CheckboxWithLabel, M_InputWithLabel } from "@/components/molecules";

import styles from "./o-form-login.module.css";

type errors = { [key: string]: string };

interface I_OFormLogin {
  onSubmit: FormEventHandler;
  errors: errors;
}

export const O_FormLogin = ({ onSubmit, errors }: I_OFormLogin) => {
  return (
    <form onSubmit={onSubmit} className={styles.o_form_login}>
      <M_InputWithLabel
        labelText="Email"
        type="email"
        name="email"
        placeholder="Informe o seu e-mail"
      />

      {errors["email"] || null}

      <M_InputWithLabel
        labelText="Senha"
        type="password"
        name="password"
        placeholder="Informe a sua senha"
      />

      {errors["password"] || null}

      <M_CheckboxWithLabel
        checked={false}
        labelText="Manter-me conectado por 30 dias."
        style={{ margin: "var(--spacing-4) 0" }}
      />

      <A_Button type="submit">Login</A_Button>
    </form>
  );
};
