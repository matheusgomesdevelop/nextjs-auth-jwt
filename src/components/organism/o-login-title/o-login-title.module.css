.o_loginTitle {
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    text-align: center;
}

.o_loginTitle > h2 {
    margin: var(--spacing-24) var(--spacing-12);
    white-space: nowrap;
}