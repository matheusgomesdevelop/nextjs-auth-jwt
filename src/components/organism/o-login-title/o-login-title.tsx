// Atoms
import { A_Avatar, A_Title, A_Text } from "@/components/atoms";

// StyleSheet
import styles from "./o-login-title.module.css";

export interface I_OLoginTitle {
  image: string;
  title: string;
  username: string;
}

export const O_LoginTitle = ({ image, title, username }: I_OLoginTitle) => {
  return (
    <div data-testid="o-login-title" className={styles.o_loginTitle}>
      <A_Avatar
        image={image}
        label={username[0]}
        size="normal"
        shape="circle"
      />
      <A_Title variant="h1" className={styles.a_title}>
        {title}
      </A_Title>
      <A_Text variant="fwReg-fs16-gray500">{username}</A_Text>
    </div>
  );
};
