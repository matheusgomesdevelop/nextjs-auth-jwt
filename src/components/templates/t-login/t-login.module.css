.t_login {
    width: 100%;
    max-width: 360px;
    display: flex;
    flex-direction: column;
    gap: var(--spacing-32);
}