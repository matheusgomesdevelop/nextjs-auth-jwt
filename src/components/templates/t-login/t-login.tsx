
// Organism and I_OTypes
import {
  I_OLoginForm,
  O_LoginForm,
} from "@/components/organism/o-login-form/o-login-form";
import {
  I_OLoginTitle,
  O_LoginTitle,
} from "@/components/organism/o-login-title/o-login-title";

// StyleSheet
import styles from "./t-login.module.css";

interface I_TLogin {
  loginTitleProps: I_OLoginTitle;
  loginFormProps: I_OLoginForm;
}

export const T_Login = ({ loginTitleProps, loginFormProps }: I_TLogin) => {
  const { image, title, username } = loginTitleProps;
  const { onSubmit, errors } = loginFormProps;

  return (
    <div data-testid="t-login" className={styles.t_login}>
      <O_LoginTitle image={image} title={title} username={username} />
      <O_LoginForm onSubmit={onSubmit} errors={errors} />
    </div>
  );
};
