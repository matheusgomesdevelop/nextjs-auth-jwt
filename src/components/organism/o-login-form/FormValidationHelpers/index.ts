import { initialValues, validationSchema } from "./login-form-validation";

// SendForm
import { sendLoginForm } from "./login-form-send";

export { initialValues, validationSchema, sendLoginForm };
