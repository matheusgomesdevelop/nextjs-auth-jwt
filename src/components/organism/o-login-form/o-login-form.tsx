import { Formik, ErrorMessage } from "formik";

// Atoms
import { A_Button } from "@/components/atoms";

// Molecules
import { M_CheckboxWithLabel, M_InputWithLabel } from "@/components/molecules";

// StyleSheet
import styles from "./o-login-form.module.css";

// Validation
import { initialValues, validationSchema } from "./FormValidationHelpers";
import { formValidation } from "./FormValidationHelpers/form-validation";

export interface I_OLoginForm {
  sendFormUrl: string;
}

export const O_LoginForm = ({ sendFormUrl }: I_OLoginForm) => {
  return (
    <Formik
      initialValues={initialValues}
      validationSchema={validationSchema}
      onSubmit={(values, { setFieldError, setSubmitting }) => {
        formValidation({ values, setFieldError, setSubmitting, sendFormUrl });
      }}
    >
      {({ getFieldProps, handleSubmit, isSubmitting }) => (
        <form onSubmit={handleSubmit} className={styles.o_form_login} data-testid="o-login-form">
          <M_InputWithLabel
            labelText="Email"
            type="email"
            placeholder="Informe o seu e-mail"
            {...getFieldProps("email")}
          />

          <ErrorMessage name="email" />

          <M_InputWithLabel
            labelText="Senha"
            type="password"
            placeholder="Informe a sua senha"
            {...getFieldProps("password")}
          />

          <ErrorMessage name="password" />

          <M_CheckboxWithLabel
            checked={false}
            labelText="Manter-me conectado por 30 dias."
            style={{ margin: "var(--spacing-4) 0" }}
          />

          <A_Button type="submit" disabled={isSubmitting}>
            Login
          </A_Button>
        </form>
      )}
    </Formik>
  );
};
