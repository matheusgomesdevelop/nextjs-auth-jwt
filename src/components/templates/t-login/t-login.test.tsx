import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

// Interfaces
import { I_OLoginForm } from "@/components/organism/o-login-form/o-login-form";
import { I_OLoginTitle } from "@/components/organism/o-login-title/o-login-title";

// Template
import { T_Login } from "./t-login";

const loginTitleProps: I_OLoginTitle = {
  image: "",
  title: "",
  username: "",
};
const loginFormUrl: I_OLoginForm = {
 sendFormUrl: "endpoint"
};

describe("Deve renderizar o t-login, corretamente", () => {
  // ==================================================
  it("Deve renderizar o o-login-title", () => {
    // Arrange
    render(<T_Login loginTitleProps={loginTitleProps} loginFormUrl={loginFormUrl} />);

    // Act
    const get_o_login_title = screen.getByTestId("o-login-title");

    // Assert
    expect(get_o_login_title).toBeInTheDocument();
  });
  // ==================================================
  it("Deve renderizar o o-login-form", () => {
    // Arrange
    render(<T_Login loginTitleProps={loginTitleProps} loginFormUrl={loginFormUrl} />);

    // Act
    const get_o_login_form = screen.getByTestId("o-login-form");

    // Assert
    expect(get_o_login_form).toBeInTheDocument();
  });

  // ==================================================
  it("Deve preservar a estrutura visual do componente", () => {
    const get_t_login = renderer
      .create(<T_Login loginTitleProps={loginTitleProps} loginFormUrl={loginFormUrl} />)
      .toJSON();

    expect(get_t_login).toMatchSnapshot();
    // ==================================================
  });
});
