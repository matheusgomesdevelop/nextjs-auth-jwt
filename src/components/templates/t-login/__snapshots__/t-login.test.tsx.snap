// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar o t-login, corretamente Deve preservar a estrutura visual do componente 1`] = `
<div
  className="t_login"
  data-testid="t-login"
  id="t-login"
>
  <div
    className="o_loginTitle"
    data-testid="o-login-title"
  >
    <div
      className="p-avatar p-component p-avatar-circle avatar"
      data-pc-name="avatar"
      data-pc-section="root"
      data-testid="a-avatar"
      style={{}}
    />
    <h1
      className="a_title"
      data-testid="a-title"
    />
    <p
      className="a_text"
      data-testid="a-text"
    />
  </div>
  <form
    className="o_form_login"
    data-testid="o-login-form"
    onSubmit={[Function]}
  >
    <div
      className="m_input_label"
      data-testid="m-input-with-label-email"
    >
      <label
        className="a_label"
        data-testid="a-label"
        htmlFor="email"
      >
        Email
      </label>
      <input
        className="p-inputtext p-component a_input"
        data-pc-name="inputtext"
        data-pc-section="root"
        data-testid="a-input"
        name="email"
        onBeforeInput={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onInput={[Function]}
        onKeyDown={[Function]}
        onPaste={[Function]}
        placeholder="Informe o seu e-mail"
        type="email"
        value=""
      />
    </div>
    <div
      className="m_input_label"
      data-testid="m-input-with-label-password"
    >
      <label
        className="a_label"
        data-testid="a-label"
        htmlFor="password"
      >
        Senha
      </label>
      <input
        className="p-inputtext p-component a_input"
        data-pc-name="inputtext"
        data-pc-section="root"
        data-testid="a-input"
        name="password"
        onBeforeInput={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onInput={[Function]}
        onKeyDown={[Function]}
        onPaste={[Function]}
        placeholder="Informe a sua senha"
        type="password"
        value=""
      />
    </div>
    <div
      className="m_checkboxWithLabel"
      data-testid="m-checkboxWithLabel"
    >
      <div
        className="p-checkbox"
        data-pc-name="checkbox"
        data-pc-section="root"
        data-testid="a-checkbox"
        id={null}
        onClick={[Function]}
        onContextMenu={null}
        onMouseDown={null}
        style={
          {
            "margin": "var(--spacing-4) 0",
          }
        }
      >
        <div
          className="p-hidden-accessible"
          data-pc-section="hiddeninputwrapper"
        >
          <input
            data-pc-section="hiddeninput"
            data-testid="a-checkbox-input"
            defaultChecked={false}
            disabled={false}
            id={null}
            name={null}
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            readOnly={false}
            required={false}
            tabIndex={null}
            type="checkbox"
          />
        </div>
        <div
          className="p-checkbox-box"
          data-p-disabled={false}
          data-p-focus={false}
          data-p-highlight={false}
          data-pc-section="input"
        />
      </div>
      <label
        className="a_label"
        data-testid="a-label"
        htmlFor="a-checkbox"
        onClick={[Function]}
      >
        Manter-me conectado por 30 dias.
      </label>
    </div>
    <button
      className="p-button p-component a_button"
      data-pc-name="button"
      data-pc-section="root"
      data-testid="a-button"
      disabled={false}
      type="submit"
    >
      Login
    </button>
  </form>
</div>
`;
